src/main/java/net/dries007/tfc/common/blocks/CharcoalPileBlock.java

```diff
@@ -0,0 +1,89 @@
+package net.dries007.tfc.common.blocks;
+
+import net.minecraft.block.*;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.IWorldReader;
+import net.minecraft.world.World;
+
+import net.dries007.tfc.client.TFCSounds;
+
+public class CharcoalPileBlock extends SnowBlock
```

Comment:
Don't override snow block, this is asking for missed override methods (and in fact, it is, and should not inherit `canBeReplaced` or `randomTick`. Also we mixin `SnowBlock`, none of which we want as part of charcoal. Neither should `instanceof SnowBlock` be true for charcoal. Just duplicate the shape + property spec.

Reply:
We love object oriented don't we folks 👍